libafl/src/bolts/mod.rs

```diff
@@ -239,73 +242,93 @@ where
     }
 }
 
+/// Stderr logger
+#[cfg(feature = "std")]
+pub static LIBAFL_STDERR_LOGGER: SimpleStderrLogger = SimpleStderrLogger::new();
+
+/// Stdout logger
+#[cfg(feature = "std")]
+pub static LIBAFL_STDOUT_LOGGER: SimpleStdoutLogger = SimpleStdoutLogger::new();
+
 /// A simple logger struct that logs to stderr when used with [`log::set_logger`].
 #[derive(Debug)]
 #[cfg(feature = "std")]
-pub struct SimpleStdErrLogger {
-    /// The min log level for which this logger will write messages.
-    pub log_level: Level,
+pub struct SimpleStdoutLogger {}
+
+impl Default for SimpleStdoutLogger {
+    fn default() -> Self {
+        Self::new()
+    }
 }
 
 #[cfg(feature = "std")]
-impl SimpleStdErrLogger {
-    /// Create a new [`log::Log`] logger that will log [`Level::Trace`] and above
+impl SimpleStdoutLogger {
+    /// Create a new [`log::Log`] logger that will wrte log to stdout
     #[must_use]
-    pub const fn trace() -> Self {
-        Self {
-            log_level: Level::Trace,
-        }
+    pub const fn new() -> Self {
+        Self {}
     }
 
-    /// Create a new [`log::Log`] logger that will log [`Level::Debug`] and above
-    #[must_use]
-    pub const fn debug() -> Self {
-        Self {
-            log_level: Level::Debug,
-        }
+    /// register stdout logger
+    pub fn register_logger() -> Result<(), Error> {
+        log::set_logger(&LIBAFL_STDOUT_LOGGER)
+            .map_err(|_| Error::unknown("Failed to register logger"))
     }
+}
 
-    /// Create a new [`log::Log`] logger that will log [`Level::Info`] and above
-    #[must_use]
-    pub const fn info() -> Self {
-        Self {
-            log_level: Level::Info,
-        }
+#[cfg(feature = "std")]
+impl log::Log for SimpleStdoutLogger {
+    #[inline]
+    fn enabled(&self, _metadata: &Metadata) -> bool {
+        true
     }
 
-    /// Create a new [`log::Log`] logger that will log [`Level::Warn`] and above
-    #[must_use]
-    pub const fn warn() -> Self {
-        Self {
-            log_level: Level::Warn,
-        }
+    fn log(&self, record: &Record) {
+        println!("{}: {}", record.level(), record.args());
     }
 
-    /// Create a new [`log::Log`] logger that will log [`Level::Error`]
+    fn flush(&self) {}
+}
+
+/// A simple logger struct that logs to stderr when used with [`log::set_logger`].
+#[derive(Debug)]
+#[cfg(feature = "std")]
+pub struct SimpleStderrLogger {}
+
+impl Default for SimpleStderrLogger {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[cfg(feature = "std")]
+impl SimpleStderrLogger {
+    /// Create a new [`log::Log`] logger that will wrte log to stdout
     #[must_use]
-    pub const fn error() -> Self {
-        Self {
-            log_level: Level::Error,
-        }
+    pub const fn new() -> Self {
+        Self {}
+    }
+
+    /// register stderr logger
+    pub fn register_logger() -> Result<(), Error> {
```

Comment:
Maybe `set_logger` since it's the name of the original API

Reply:
ok